framework/src/bcs/NodalBC.C

```diff
@@ -43,6 +77,13 @@ NodalBC::computeResidual(NumericVector<Number> & residual)
     dof_id_type & dof_idx = _var.nodalDofIndex();
     _qp = 0;
     residual.set(dof_idx, computeQpResidual());
+
+    if (_has_save_in)
+    {
+      Threads::spin_mutex::scoped_lock lock(Threads::spin_mtx);
+      for (unsigned int i=0; i<_save_in.size(); i++)
+        _save_in[i]->sys().solution().set(dof_idx, 0);
```

Comment:
This doesn't quite work because other things that do `save_in` use `solution().add()` and you can't mix `add()` and `set()` without a `close()`.

You could fix that up by putting an AuxiliarySystem solution `close()` at the beginning of `NonlinearSystem::computeNodalBCs()`... and then another at the end.  However... we only want to do that when someone is using NodalBC `save_in`.

Further, if you're going to set something then you need to set the actual residual value... not zero.